src/netaddress.cpp

```diff
@@ -271,6 +275,30 @@ bool CNetAddr::SetSpecial(const std::string& str)
     return false;
 }
 
+bool CNetAddr::SetI2P(const std::string& str)
+{
+    static const char* suffix{".b32.i2p"};
```

Comment:
It seems I2P software support addresses that use uppercase characters but they make sure to convert them to lowercase before processing them everywhere. Take a look at this for example: https://github.com/i2p/i2p.i2p/blob/9ba86e86aa1f320a9ddbb644e09afcdf0d1b01ba/apps/i2ptunnel/java/src/net/i2p/i2ptunnel/I2PTunnelHTTPClient.java#L814

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/20685, thanks!